src/Ilios/AuthenticationBundle/RelationshipVoter/ApplicationConfig.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Ilios\AuthenticationBundle\RelationshipVoter;
+
+use Ilios\AuthenticationBundle\Classes\SessionUserInterface;
+use Ilios\CoreBundle\Entity\ApplicationConfigInterface;
+use Ilios\CoreBundle\Entity\DTO\ApplicationConfigDTO;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+class ApplicationConfig extends AbstractVoter
+{
+    protected function supports($attribute, $subject)
+    {
+        return (
+            ($subject instanceof ApplicationConfigDTO && in_array($attribute, [self::VIEW])) or
+            ($subject instanceof ApplicationConfigInterface && in_array($attribute, [
+                    self::CREATE, self::VIEW, self::EDIT, self::DELETE
+                ]))
+        );
+    }
+
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
+    {
+        $user = $token->getUser();
+        if (!$user instanceof SessionUserInterface) {
+            return false;
+        }
+        if ($user->isRoot()) {
+            return true;
+        }
+
+        if ($subject instanceof ApplicationConfigDTO) {
```

Comment:
Only root should be able to read the application config. It contains sensitive info like the ldap bind password.

Reply:
fixed.